main/src/ca/uwaterloo/flix/language/ast/NamedAst.scala

```diff
@@ -49,7 +49,9 @@ object NamedAst {
 
     case class Enum(doc: Ast.Doc, ann: Ast.Annotations, mod: Ast.Modifiers, sym: Symbol.EnumSym, tparams: List[TypeParam], derives: Derivations, cases: List[Declaration.Case], loc: SourceLocation) extends Declaration
 
-    case class Struct(doc: Ast.Doc, ann: Ast.Annotations, mod: Ast.Modifiers, sym: Symbol.StructSym, tparams: List[TypeParam], fields: List[StructField], loc: SourceLocation) extends Declaration
+    case class Struct(doc: Ast.Doc, ann: Ast.Annotations, mod: Ast.Modifiers, sym: Symbol.StructSym, tparams: List[TypeParam], fields: List[StructField], indices: Map[Name.Label, Int], loc: SourceLocation) extends Declaration
+
+    case class StructField(sym: Name.Label, tpe: Type, loc: SourceLocation)
```

Comment:
I would move this towards the bottom of the file.

For some reason `Case` is also here, but neither `StructField` nor `Case` are declarations (they are declaration fragments).

Reply:
But I now see Case extends declaration, but this makes no sense to me.

Reply:
For lack of a better idea, let us leave this for now.

Reply:
I think it's because in `Resolver` there are functions like `infalliableLookupSym` that expect a return a declaration given a symbol lookup and we wanted to include these symbols there